source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -16,12 +16,16 @@ namespace Extensions {
 namespace HttpFilters {
 namespace HeaderToMetadataFilter {
 
-Config::Config(const envoy::extensions::filters::http::header_to_metadata::v3::Config config) {
+Config::Config(const envoy::extensions::filters::http::header_to_metadata::v3::Config config,
+               const bool per_route) {
   request_set_ = Config::configToVector(config.request_rules(), request_rules_);
   response_set_ = Config::configToVector(config.response_rules(), response_rules_);
 
-  // don't allow an empty configuration
-  if (!response_set_ && !request_set_) {
+  // Note: empty configs are fine for the global config, which would be the case for enabling
+  //       the filter globally without rules and then applying them at the virtual host or
+  //       route level. At the virtual or route level, it makes no sense to have an empty
+  //       config so we throw an error.
+  if (per_route && !response_set_ && !request_set_) {
     throw EnvoyException("Must at least specify either response or request config");
```

Comment:
while you're in here: `Must specify at least response or request config`? maybe even specify whether its per route or global so people know what to fix

Reply:
Ah yes.